bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -0,0 +1,203 @@
+        logger.debug("Sending long poll request");
+
+        JsonRpcRequest requestContent = new JsonRpcRequest("2.0", "RE/longPoll", new String[] { subscriptionId, "20" });
+        String url = httpClient.createUrl("remote/json-rpc");
+        Request request = httpClient.createRequest(url, POST, requestContent);
+
+        // Long polling responds after 20 seconds with an empty response if no update has happened.
+        // 10 second threshold was added to not time out if response from controller takes a bit longer than 20 seconds.
+        request.timeout(30, TimeUnit.SECONDS);
+
+        this.request = request;
+        LongPolling longPolling = this;
+        request.send(new BufferingResponseListener() {
+            @Override
+            public void onComplete(@Nullable Result result) {
+                Throwable failure = result != null ? result.getFailure() : null;
+                if (failure != null) {
+                    if (failure instanceof ExecutionException) {
+                        if (failure.getCause() instanceof AbortLongPolling) {
+                            logger.debug("Canceling long polling for subscription id {} because it was aborted",
+                                    subscriptionId);
+                        } else {
+                            longPolling.handleFailure.accept(new LongPollingFailedException(
+                                    "Unexpected exception during long polling request", failure));
+                        }
+                    } else {
+                        longPolling.handleFailure.accept(new LongPollingFailedException(
+                                "Unexpected exception during long polling request", failure));
+                    }
+                } else {
+                    longPolling.onLongPollResponse(httpClient, subscriptionId, this.getContentAsString());
+                }
+            }
+        });
+    }
+
+    private void onLongPollResponse(BoschHttpClient httpClient, String subscriptionId, String content) {
+        // Check if thing is still online
+        if (this.aborted) {
+            logger.debug("Canceling long polling for subscription id {} because it was aborted", subscriptionId);
+            return;
+        }
+
+        logger.debug("Long poll response: {}", content);
+
+        String nextSubscriptionId = subscriptionId;
+
+        LongPollResult longPollResult = gson.fromJson(content, LongPollResult.class);
+        if (longPollResult != null) {
+            this.handleResult.accept(longPollResult);
+        } else {
+            logger.warn("Could not parse long poll response: {}", content);
+
+            // Check if we got a proper result from the SHC
+            LongPollError longPollError = gson.fromJson(content, LongPollError.class);
+
+            if (longPollError != null) {
+                logger.warn("Got error from SHC: {}", longPollError.error.hashCode());
```

Comment:
I don't think logging the hashcode is going to very helpful for debugging...

Reply:
Oops, logging the message and code now :)